OpenRA.Mods.RA2/Orders/EnterGarrisonOrderTargeter.cs

```diff
@@ -0,0 +1,89 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.RA2.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Orders
+{
+    public class EnterGarrisonOrderTargeter<GarrisonableInfo> : UnitOrderTargeter where GarrisonableInfo : ITraitInfo
+    {
+        readonly AlternateGarrisonMode mode;
+        readonly Func<Actor, Actor, bool> canTarget;
+        readonly Func<Actor, Actor, bool> useEnterCursor;
+
+        public EnterGarrisonOrderTargeter(string order, int priority,
+            Func<Actor, Actor, bool> canTarget, Func<Actor, Actor, bool> useEnterCursor, AlternateGarrisonMode mode)
+            : base(order, priority, "enter", true, true)
+        {
+            this.canTarget = canTarget;
+            this.useEnterCursor = useEnterCursor;
+            this.mode = mode;
+        }
+
+        public override bool CanTargetActor(Actor self, Actor target, TargetModifiers modifiers, ref string cursor)
+        {
+            switch (mode)
+            {
+                case AlternateGarrisonMode.None:
+                    return false;
+                case AlternateGarrisonMode.Force:
+                    if (!modifiers.HasModifier(TargetModifiers.ForceMove))
+                        return false;
+                    break;
+                case AlternateGarrisonMode.Default:
+                    if (modifiers.HasModifier(TargetModifiers.ForceMove))
+                        return false;
+                    break;
+                case AlternateGarrisonMode.Always:
+                    break;
+            }
+
+            if ((target.Owner.PlayerName == "Creeps" || target.Owner.PlayerName == "Neutral" || self.Owner.IsAlliedWith(target.Owner)) && target.Info.HasTraitInfo<GarrisonableInfo>())
```

Comment:
This should use alliance check and pass on Stance.Neutral. Creeps are enemy, they shouldn't be allowed!

Reply:
Thank you for the info. I actually had no idea what creeps were. Are creeps basically like Neutral except they shoot you?

Reply:
Even the hatrdcoded Neutral lookup is a bug to me. `target.Owner.HasStance(Stance.Neutral)` should substitute.

Reply:
Yes, Creeps are4 the hostile neutrals. Like TS/TD Visceroids.

Reply:
A better option would be that this directly uses Garrisoner's stances as well.

Reply:
target.Owner.HasStance(Stance.Neutral)
target.Owner doesn't have a method called HasStance. closest thing I can find is in the info class. Which you also commented on. I'm not sure I understand how stances work or what they are.
https://github.com/MustaphaTR/Romanovs-Vengeance/pull/13/files#diff-24d7408fd3ae598f11a5598df88c406cR43

Reply:
`garrisonableInfo.TargetStances.HasStance(target.Owner.Stances(self.Owner))`

A Stance can have the value of Ally, Neutral or Enemy and defines the diplomatic stances between two players.